File: OC/TradePlatform/Class/Account(账户)/AloneInfo(个人信息)/View/AloneInfoView.m
//
//  AloneInfoView.m
//  TradePlatform
//
//  Created by apple on 2017/1/3.
//  Copyright © 2017年 apple. All rights reserved.
//

#import "AloneInfoView.h"

@implementation AloneInfoView

- (instancetype)init
{
    self = [super init];
    if (self) {
        [self aloneInfoLayoutView];
    }
    return self;
}

- (void)aloneInfoLayoutView {
    /** 名字 */
    self.accountName = [[CustomCell alloc] init];
    self.accountName.lineStyle = NotLine;
    self.accountName.cellStyle = HorizontalLayoutNotMImgAndVImg;
    self.accountName.mainLabel.text = @"姓名";
    self.accountName.mainLabel.font = FifteenTypeface;
    self.accountName.mainLabel.textColor = Black;
    self.accountName.rightViceLabel.font = ThirteenTypeface;
    self.accountName.rightViceLabel.textColor = GrayH1;
    self.accountName.mainBtn.tag = AccountNameBtnAction;
    [self addSubview:self.accountName];
    /** 手机号 */
    self.telPhone = [[CustomCell alloc] init];
    self.telPhone.lineStyle = NotLine;
    self.telPhone.cellStyle = HorizontalLayoutNotMImgAndVImg;
    self.telPhone.mainLabel.text = @"手机号";
    self.telPhone.mainLabel.font = FifteenTypeface;
    self.telPhone.mainLabel.textColor = Black;
    self.telPhone.rightViceLabel.font = ThirteenTypeface;
    self.telPhone.rightViceLabel.textColor = GrayH1;
    self.telPhone.mainBtn.tag = TelPhoneBtnAction;
    [self addSubview:self.telPhone];
    /** 修改密码 */
    self.delPassword = [[CustomCell alloc] init];
    self.delPassword.lineStyle = NotLine;
    self.delPassword.cellStyle = HorizontalLayoutNotMImgAndHaveVImgAndNotVBtn;
    self.delPassword.mainLabel.text = @"修改密码";
    self.delPassword.mainLabel.font = FifteenTypeface;
    self.delPassword.mainLabel.textColor = Black;
    self.delPassword.leftViceLabel.font = ThirteenTypeface;
    self.delPassword.leftViceLabel.textColor = GrayH1;
    self.delPassword.mainBtn.tag = DelPasswordBtnAction;
    [self addSubview:self.delPassword];
}

- (void)layoutSubviews {
    [super layoutSubviews];
    @weakify(self)
    /** 名字 */
    [self.accountName mas_makeConstraints:^(MASConstraintMaker *make) {
        @strongify(self)
        make.left.equalTo(self.mas_left);
        make.right.equalTo(self.mas_right);
        make.top.equalTo(self.mas_top).offset(10);
        make.height.mas_equalTo(@50);
    }];
    /** 手机号 */
    [self.telPhone mas_makeConstraints:^(MASConstraintMaker *make) {
        @strongify(self)
        make.left.equalTo(self.mas_left);
        make.right.equalTo(self.mas_right);
        make.top.equalTo(self.accountName.mas_bottom).offset(10);
        make.height.mas_equalTo(@50);
    }];
    /** 修改密码 */
    [self.delPassword mas_makeConstraints:^(MASConstraintMaker *make) {
        @strongify(self)
        make.left.equalTo(self.mas_left);
        make.right.equalTo(self.mas_right);
        make.top.equalTo(self.telPhone.mas_bottom).offset(10);
        make.height.mas_equalTo(@50);
    }];
}

@end

File: Swict/LoveCarArtisan/Class/Account(账户)/AloneInfo(个人信息)/View/AloneInfoView.swift
//
//  AloneInfoView.swift
//  LoveCarArtisan
//
//  Created by apple on 2017/5/19.
//  Copyright © 2017年 apple. All rights reserved.
//

import UIKit

class AloneInfoView: UIView {
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
    override init(frame: CGRect) {
        super.init(frame: frame)
        aloneInfolaoutView()
    }
    // MARK: 创建控件
    /** 账户名view */
    lazy var accountName: CustomCell = {
        let view = CustomCell()
        view.cellStyle = CustomCellStyle.HorizontalLayoutNotMImgAndVImg
        view.lineStyle = DividingLineStyle.NotLine
        view.mainLabel.text = "账户名"
        view.mainLabel.font = FifteenFont
        return view
    }()
    /** 账户手机号view */
    lazy var accountPhone: CustomCell = {
        let view = CustomCell()
        view.cellStyle = CustomCellStyle.HorizontalLayoutNotMImgAndVImg
        view.lineStyle = DividingLineStyle.NotLine
        view.mainLabel.text = "账户手机号"
        view.mainLabel.font = FifteenFont
        return view
    }()
    /** 修改密码view */
    lazy var delPassword: CustomCell = {
        let view = CustomCell()
        view.cellStyle = CustomCellStyle.HorizontalLayoutNotMImgAndHaveVImgAndNotVBtn
        view.lineStyle = DividingLineStyle.NotLine
        view.mainLabel.text = "修改密码"
        view.mainLabel.font = FifteenFont
        return view
    }()
    // MARK: 视图布局
    func aloneInfolaoutView() {
        /** 账户名view */
        addSubview(accountName)
        /** 账户手机号view */
        addSubview(accountPhone)
        /** 修改密码view */
        addSubview(delPassword)
    }
    
    override func layoutSubviews() {
        super.layoutSubviews()
        /** 账户名view */
        accountName.mas_makeConstraints { (make: MASConstraintMaker!) in
            make.top.equalTo()(self.mas_top)?.setOffset(10)
            make.left.equalTo()(self.mas_left)
            make.right.equalTo()(self.mas_right)
            make.height.mas_equalTo()(50)
        }
        /** 账户手机号view */
        accountPhone.mas_makeConstraints { (make: MASConstraintMaker!) in
            make.top.equalTo()(self.accountName.mas_bottom)?.setOffset(10)
            make.left.equalTo()(self.mas_left)
            make.right.equalTo()(self.mas_right)
            make.height.mas_equalTo()(50)
        }
        /** 修改密码view */
        delPassword.mas_makeConstraints { (make: MASConstraintMaker!) in
            make.top.equalTo()(self.accountPhone.mas_bottom)?.setOffset(10)
            make.left.equalTo()(self.mas_left)
            make.right.equalTo()(self.mas_right)
            make.height.mas_equalTo()(50)
        }
    }
}
